core/src/main/java/hivemall/math/matrix/MatrixUtils.java

```diff
@@ -70,4 +77,259 @@ public void apply(int i, int value) {
+        Preconditions.checkArgument(rows.length == nnz);
+        Preconditions.checkArgument(cols.length == nnz);
+
+        final int[] columnPointers = new int[numCols + 1];
+        final int[] rowIndicies = new int[nnz];
+        final float[] values = new float[nnz];
+
+        coo2csr(cols, rows, data, columnPointers, rowIndicies, values, numCols, numRows, nnz);
+
+        if (sortRows) {
+            sortIndicies(columnPointers, rowIndicies, values);
+        }
+
+        return new CSCFloatMatrix(columnPointers, rowIndicies, values, numRows, numCols);
+    }
+
+    private static void coo2csr(@Nonnull final int[] rows, @Nonnull final int[] cols,
+            @Nonnull final double[] data, @Nonnull final int[] rowPointers,
+            @Nonnull final int[] colIndicies, @Nonnull final double[] values,
+            @Nonnegative final int numRows, @Nonnegative final int numCols, final int nnz) {
+        // compute nnz per for each row to get rowPointers
+        for (int n = 0; n < nnz; n++) {
+            rowPointers[rows[n]]++;
+        }
+        for (int i = 0, sum = 0; i < numRows; i++) {
+            int curr = rowPointers[i];
+            rowPointers[i] = sum;
+            sum += curr;
+        }
+        rowPointers[numRows] = nnz;
+
+        // copy cols, data to colIndicies, csrValues
+        for (int n = 0; n < nnz; n++) {
+            int row = rows[n];
+            int dst = rowPointers[row];
+
+            colIndicies[dst] = cols[n];
+            values[dst] = data[n];
+
+            rowPointers[row]++;
+        }
+
+        for (int i = 0, last = 0; i <= numRows; i++) {
+            int tmp = rowPointers[i];
+            rowPointers[i] = last;
+            last = tmp;
+        }
+    }
+
+    private static void coo2csr(@Nonnull final int[] rows, @Nonnull final int[] cols,
+            @Nonnull final float[] data, @Nonnull final int[] rowPointers,
+            @Nonnull final int[] colIndicies, @Nonnull final float[] values,
+            @Nonnegative final int numRows, @Nonnegative final int numCols, final int nnz) {
+        // compute nnz per for each row to get rowPointers
+        for (int n = 0; n < nnz; n++) {
+            rowPointers[rows[n]]++;
+        }
+        for (int i = 0, sum = 0; i < numRows; i++) {
+            int curr = rowPointers[i];
+            rowPointers[i] = sum;
+            sum += curr;
+        }
+        rowPointers[numRows] = nnz;
+
+        // copy cols, data to colIndicies, csrValues
+        for (int n = 0; n < nnz; n++) {
+            int row = rows[n];
+            int dst = rowPointers[row];
+
+            colIndicies[dst] = cols[n];
+            values[dst] = data[n];
+
+            rowPointers[row]++;
+        }
+
+        for (int i = 0, last = 0; i <= numRows; i++) {
+            int tmp = rowPointers[i];
+            rowPointers[i] = last;
+            last = tmp;
+        }
+    }
+
+    private static void sortIndicies(@Nonnull final int[] rowPointers,
```

Comment:
Since this method is used both for CSR and CSC, more generic argument names make the code more understandable e.g., `majorAxisPointers` and `minorAxisIndices`